cpp/include/cuml/fil/multi_reduction.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#include <raft/cuda_utils.cuh>
+// data [T]ype, reduction [R]adix
+template <int R, typename T>
+/**
+ per_thread holds one value per thread in shared memory
```

Comment:
The documentation comment should also describe what the function does, the assumptions made (e.g. `n_groups * n_values < number_of_threads_in_block`) and side effects (`data` is modified during reduction).